pkg/collector/corechecks/snmp/checkconfig/config.go

```diff
@@ -211,12 +239,42 @@ func NewCheckConfig(rawInstance integration.Data, rawInitConfig integration.Data
 		c.ExtraTags = strings.Split(instance.ExtraTags, ",")
 	}
 
-	if instance.Network != "" {
-		log.Warnf("`network_address` config is not available for corecheck SNMP integration to use autodiscovery. Agent `snmp_listener` config can be used instead: https://docs.datadoghq.com/network_monitoring/devices/setup?tab=snmpv2#autodiscovery")
+	c.Network = instance.Network
+	if instance.DiscoveryWorkers == 0 {
+		c.DiscoveryWorkers = defaultDiscoveryWorkers
+	} else {
+		c.DiscoveryWorkers = instance.DiscoveryWorkers
+	}
+
+	if instance.Workers == 0 {
+		c.Workers = defaultWorkers
+	} else {
+		c.Workers = instance.Workers
 	}
 
-	if c.IPAddress == "" {
-		return nil, fmt.Errorf("ip_address config must be provided")
+	if instance.DiscoveryAllowedFailures == 0 {
+		c.DiscoveryAllowedFailures = defaultDiscoveryAllowedFailures
+	} else {
+		c.DiscoveryAllowedFailures = instance.DiscoveryAllowedFailures
+	}
+
+	if instance.DiscoveryInterval == 0 {
+		c.DiscoveryInterval = defaultDiscoveryInterval
+	} else {
+		c.DiscoveryInterval = instance.DiscoveryInterval
+	}
+
+	c.IgnoredIPAddresses = make(map[string]bool, len(instance.IgnoredIPAddresses))
+	for _, ipAddress := range instance.IgnoredIPAddresses {
+		c.IgnoredIPAddresses[ipAddress] = true
+	}
+
+	if c.IPAddress == "" && c.Network == "" {
```

Comment:
What about early checking for these 2 conditions and have early returns? No need to copy all the other params if the configuration is invalid

Reply:
👍 , move here cf95f4a0a